src/Mapper/ModelPropertyMapper.cs

```diff
@@ -120,7 +120,24 @@ public ClientModelProperty Map(PropertyJv property)
 
             bool wasFlattened = property.WasFlattened();
 
-            return new ClientModelProperty(property.Name, description, annotationArguments, isXmlAttribute, xmlName, serializedName, isXmlWrapper, xmlListElementName, propertyWireType, propertyClientType, isConstant, defaultValue, isReadOnly, wasFlattened, headerCollectionPrefix);
+            bool useVarArgs = false;
+
+            if (property.Parent is CompositeType parent)
+            {
+                var modelPropertyName= $"{parent.Name}.{property.Name}";
+                if (Settings.Instance.CustomSettings["ModelProperties"] is Dictionary<string, Dictionary<string, bool>> modelProperties)
```

Comment:
this if check seems redundant because ProcessInternal() in Program always sets it to a dictionary